http-upload-server/rest/upload.go

```diff
@@ -0,0 +1,73 @@
+package rest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+func (h *Handlers) Upload(w http.ResponseWriter, r *http.Request) {
+	srcFile, _, err := r.FormFile("file")
+	if err != nil {
+		h.log.Error("failed to acquire the uploaded content", zap.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		writeError(w, err)
+		return
+	}
+	defer srcFile.Close()
+	size, err := getSize(srcFile)
+	if err != nil {
+		h.log.Error("failed to get the size of the uploaded content", zap.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		writeError(w, err)
+		return
+	}
+
+	body, err := ioutil.ReadAll(srcFile)
+	if err != nil {
+		h.log.Error("failed to read the uploaded content", zap.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		writeError(w, err)
+		return
+	}
+	filename := uuid.New().String()
+
+	dstPath := path.Join(h.uploadFolder, filename)
+	dstFile, err := os.OpenFile(dstPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		h.log.Error("failed to open the file", zap.Error(err), zap.String("path", dstPath))
+		w.WriteHeader(http.StatusInternalServerError)
+		writeError(w, err)
+		return
+	}
+	defer dstFile.Close()
+	if written, err := dstFile.Write(body); err != nil {
```

Comment:
on systems where int is alias to int32 this will fail to upload files larger than max int32